### ISSUE TITLE
PolynomialCFD Crossing Calculation

### ISSUE THREAD

**Issue by ksmith0** [pixie16/paass#164]
@spaulaus and I had some discussion about the PolynomialCFD crossing determination.

In PolynomialCfd.cpp the following is used to determine the phase.
```c++
// Calculate the phase of the trace.
phase = (-result[1] + 
   sqrt(result[1] * result[1] - 4 * result[2] * (result[0] - threshold))) /
   (2 * result[2]);
```
We believe, the plus sign between `-b` and `sqrt(b^2-4ac)` is required if `a > 0`, but should be a negative sign for `a < 0`.
 
In addition, we are not sure why the 'c' term in the '4ac' term is `(result[0] - threshold)`, we believe it should be `result[0]`.

**Comment by ksmith0** [comment]
@cthornsb What do you think about this?

**Comment by ksmith0** [comment]
The 'c' term should be `result[0] - threshold` as we are solving for the x value at the threshold not at 0 as is typically done with the quadratic equation.

**Comment by cthornsb** [comment]
And yes, I agree that you should take the negative solution for downward opening parabolas. But that condition is a < 0, not b < 0 correct?

**Comment by ksmith0** [comment]
You are correct it should be a < 0. I have revised the original comment to reflect this.

**Comment by spaulaus** [comment]
Addressed in 1beb17e90e86dd4893c08b8ac768bbb2d213ed92
